tests/test_pipreqs.py

```diff
@@ -109,6 +115,16 @@ def test_get_use_local_only(self):
         for item in imports_with_info:
             self.assertTrue(item['name'].lower() in self.local)
 
+    def test_get_locally_installed_packages(self):
+        with patch.object(sys, 'path', new=[self.local_packages_path]):
+            local_installed_packages = pipreqs.get_locally_installed_packages()
+            expected_output = [
+                {'exports': [], 'name': 'invalid_package', 'version': None},
+                {'exports': ['valid_package'], 'name': 'valid_package', 'version': '1.0.0'},
+                {'exports': [], 'name': 'EGG', 'version': 'INFO'}
```

Comment:
sorry, I dont understand this test. Could you explain it to me? why this becomes the output and why is the module invalid? Is it because it haz an additional folder?

Reply:
First of all, we based our tests on our understanding of the code. We don't know exactly how pip builds the structure of a package.

1. The first test case was not necessary, so we decided to remove it.
2. The second one was called as a "valid_package" because its "top_level.txt" file has one module named "valid_package" that is not in our code's blacklist (which is a variable called ignore with value `["tests", "_tests", "egg", "EGG", "info"]`) and also has a module named "tests" that is in that blacklist.
3. The last test case makes sure that "top_level.txt" files that are inside a folder that starts with "EGG" are not considered.

We tried to understand the code's behavior by reading "setuptools" documentation ([link](https://setuptools.pypa.io/en/latest/deprecated/python_eggs.html)), but we did not understand much. It requires a lot of reading and time for understanding.

If you think it's okay for us to use our time to study this and then validate if it's right, that is okay.